gobblin-hive-registration/src/main/java/gobblin/hive/HiveRegistrationUnit.java

```diff
@@ -338,6 +351,124 @@ public void setSerDeType(String serDeType) {
     this.serDeType = Optional.of(serDeType);
   }
 
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    result = prime * result + ((this.bucketColumns == null) ? 0 : this.bucketColumns.hashCode());
+    result = prime * result + ((this.columns == null) ? 0 : this.columns.hashCode());
+    result = prime * result + ((this.dbName == null) ? 0 : this.dbName.hashCode());
+    result = prime * result + ((this.inputFormat == null) ? 0 : this.inputFormat.hashCode());
+    result = prime * result + ((this.isCompressed == null) ? 0 : this.isCompressed.hashCode());
+    result = prime * result + ((this.isStoredAsSubDirs == null) ? 0 : this.isStoredAsSubDirs.hashCode());
+    result = prime * result + ((this.location == null) ? 0 : this.location.hashCode());
+    result = prime * result + ((this.numBuckets == null) ? 0 : this.numBuckets.hashCode());
+    result = prime * result + ((this.outputFormat == null) ? 0 : this.outputFormat.hashCode());
+    result = prime * result + ((this.serDeType == null) ? 0 : this.serDeType.hashCode());
+    result = prime * result + ((this.tableName == null) ? 0 : this.tableName.hashCode());
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null) {
+      return false;
+    }
+    if (getClass() != obj.getClass()) {
+      return false;
+    }
+    HiveRegistrationUnit other = (HiveRegistrationUnit) obj;
+    if (this.bucketColumns == null) {
+      if (other.bucketColumns != null) {
+        return false;
+      }
+    } else if (!this.bucketColumns.equals(other.bucketColumns)) {
+      return false;
+    }
+    if (this.columns == null) {
```

Comment:
What if `this.columns != null` and `other.columns == null`? Similar issue with the rest comparison block.